github/resource_github_organization_ruleset_test.go

```diff
@@ -597,22 +597,40 @@ func TestOrganizationPushRulesetSupport(t *testing.T) {
 	// Test expand functionality (organization rulesets use org=true)
 	expandedRules := expandRules(input, true)
 
-	if len(expandedRules) != 4 {
-		t.Fatalf("Expected 4 expanded rules for organization push ruleset, got %d", len(expandedRules))
+	if expandedRules == nil {
+		t.Fatal("Expected expanded rules to not be nil")
 	}
 
 	// Verify we have all expected push rule types
-	ruleTypes := make(map[string]bool)
-	for _, rule := range expandedRules {
-		ruleTypes[rule.Type] = true
+	ruleCount := 0
+	if expandedRules.FilePathRestriction != nil {
+		ruleCount++
+		if len(expandedRules.FilePathRestriction.RestrictedFilePaths) != 3 {
+			t.Errorf("Expected 3 restricted file paths, got %d", len(expandedRules.FilePathRestriction.RestrictedFilePaths))
+		}
 	}
-
-	expectedPushRules := []string{"file_path_restriction", "max_file_size", "max_file_path_length", "file_extension_restriction"}
-	for _, expectedType := range expectedPushRules {
-		if !ruleTypes[expectedType] {
-			t.Errorf("Expected organization push rule type %s not found in expanded rules", expectedType)
+	if expandedRules.MaxFileSize != nil {
+		ruleCount++
+		if expandedRules.MaxFileSize.MaxFileSize != 100 {
+			t.Errorf("Expected max file size to be 100, got %d", expandedRules.MaxFileSize.MaxFileSize)
 		}
 	}
+	if expandedRules.MaxFilePathLength != nil {
+		ruleCount++
+		if expandedRules.MaxFilePathLength.MaxFilePathLength != 250 {
+			t.Errorf("Expected max file path length to be 250, got %d", expandedRules.MaxFilePathLength.MaxFilePathLength)
+		}
+	}
+	if expandedRules.FileExtensionRestriction != nil {
+		ruleCount++
+		if len(expandedRules.FileExtensionRestriction.RestrictedFileExtensions) != 4 {
```

Comment:
question: Same for these conditionals. Would it make sense to refer the `rulesMap` values directly instead of hard-coding them?

Reply:
@nickfloyd it's worth noting that the duplication of the expected value in the test and the error message caused quite a lot of overhead in the patch alignment PR. I'm also not sure I agree that this can be classed as an idiom and would be tempted to follow the boy scout rule and fix it as part of the changes.

Reply:
Absolutely. I can totally get behind that.  My concern is that this PR is already huge - so I can scope the modifications to just this file and any others related to it - we should follow up with a cleaner approach across the entire provider.

Reply:
Yeah... I feel like there might be a better (cleaner) way to handle things like that - orthogonal to this change but I'd like to include it in our patterns handbook (when we write one 😉)